packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -756,134 +773,192 @@ describe("<FilterBar />", () => {
   })
 
   describe("Context use cases", () => {
-    type Items = Array<{ value: string; label: string }>
-
-    type AsyncValues = {
-      city: string[]
-      hero: string[]
-    }
-
-    const MockFilterAsyncComponent = ({
-      id,
-      fetcher,
-    }: {
-      id: string
-      fetcher: (args: Partial<AsyncValues>) => Promise<Items>
-    }): JSX.Element => {
-      const [items, setItems] = useState<Items>([])
-      const { getActiveFilterValues } = useFilterBarContext()
-      const activeFilterVals = getActiveFilterValues()
-
-      useEffect(() => {
-        fetcher(activeFilterVals).then(fetchedItems => {
-          if (JSON.stringify(fetchedItems) !== JSON.stringify(items)) {
-            setItems(fetchedItems)
-          }
-        })
-      }, [JSON.stringify(activeFilterVals)])
-
-      return (
-        <FilterBar.MultiSelect id={id} items={items}>
-          {() => (
-            <FilterMultiSelect.ListBox>
-              {({ allItems }) => (
-                <FilterMultiSelect.ListBoxSection
-                  items={allItems}
-                  sectionName="All Items"
-                >
-                  {item => (
-                    <FilterMultiSelect.Option key={item.key} item={item} />
-                  )}
-                </FilterMultiSelect.ListBoxSection>
-              )}
-            </FilterMultiSelect.ListBox>
-          )}
-        </FilterBar.MultiSelect>
-      )
-    }
-
-    const fetchCityOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
-      const isSupermanInFilterValue = filterValues.hero?.includes("superman")
-      const isBatmanInFilterValue = filterValues.hero?.includes("batman")
+    describe("getActiveFilterValues()", () => {
+      type Items = Array<{ value: string; label: string }>
 
-      if (isBatmanInFilterValue && !isSupermanInFilterValue) {
-        return Promise.resolve([{ value: "gotham", label: "Gotham" }])
+      type AsyncValues = {
+        city: string[]
+        hero: string[]
       }
 
-      return Promise.resolve([
-        { value: "gotham", label: "Gotham" },
-        { value: "metro", label: "Metropolis" },
-      ])
-    })
-
-    const fetchHeroOptions = jest.fn((filterValues: Partial<AsyncValues>) => {
-      const isGothamInFilterValue = filterValues.city?.includes("gotham")
-      const isMetroInFilterValue = filterValues.city?.includes("metro")
+      const MockFilterAsyncComponent = ({
```

Comment:
Doe we need to have an async component to prove out this test...? Concerned we're making our tests too close to specific use cases.